### PR TITLE
Add DigitalOcean "one-click-install" button

### PR DESCRIPTION
**Pull request by BrunoBernardino**
This adds the button to the README and fixes the deploy template to actually work (still to be tested).

URL to test: https://cloud.digitalocean.com/apps/new?repo=https://github.com/padloc/padloc/tree/feature-digitalocean-one-click-install (I had to do this PR instead of https://github.com/padloc/padloc/pull/636 because the `/` in the branch name broke DO...)

### CODE REVIEW

**Comment by BrunoBernardino** [comment]
@MaKleSoft I'm thinking of adding an env var to disable CSP, since that makes the initial setup harder and isn't necessarily increasing the security in those situations. It causes a problem using this as well, so I'd do it as part of this PR. What do you think?

**Comment by MaKleSoft** [comment]
> I'm thinking of adding an env var to disable CSP, since that makes the initial setup harder and isn't necessarily increasing the security in those situations. It causes a problem using this as well, so I'd do it as part of this PR. What do you think?

Sure, why not! As long as it's optional.

File: .do/deploy.template.yaml [hunk 1]
```diff
@@ -1,81 +1,49 @@
...
+          scope: RUN_AND_BUILD_TIME
+          value: ${server.PUBLIC_URL}
+        - key: PL_PWA_URL
+          scope: RUN_AND_BUILD_TIME
+          value: ${pwa.PUBLIC_URL}
+    services:
+        - build_command: npm ci
+          environment_slug: node-js
+          git:
+              branch: feature-digitalocean-one-click-install
```
Comment (BrunoBernardino) [inline .do/deploy.template.yaml:26]: This needs to be reverted to `main` before merging this, but it's necessary to keep testing.

File: .do/deploy.template.yaml [hunk 2]
```diff
@@ -1,81 +1,49 @@
...
+          health_check:
+              http_path: /healthcheck
+          run_command: cd packages/server && npm start
+          source_dir: /
+    static_sites:
+        - build_command: npm ci && cd packages/pwa && npm run build
+          catchall_document: index.html
+          environment_slug: node-js
+          git:
+              branch: feature-digitalocean-one-click-install
```
Comment (BrunoBernardino) [inline .do/deploy.template.yaml:43]: This needs to be reverted to `main` before merging this, but it's necessary to keep testing.

**Comment by BrunoBernardino** [comment]
The app now installs and loads, but doesn't get through the verification code (it validates, but keeps requesting it). I'll need to investigate to see what's wrong/missing, but there was no error in the logs. I suspect something with the automatically-built URL env vars (including/not-including a trailing slash, for example). Thoughts welcome if you have any @MaKleSoft

**Comment by BrunoBernardino** [comment]
@MaKleSoft I'm still getting an issue about the verification code not matching: https://github.com/padloc/padloc/blob/98a73dc43aee9c40347cc25e6071801c3c4c7bc6/packages/core/src/auth/email.ts#L61-L73

Do you think it can be something to do with the fact we're using `leveldb` here? I'm not 100% sure if the App Platform has the instances with the disks in write mode. If you're not sure I'll try setting it up to a `postgres` db, but that's unnecessarily expensive for small/personal use. Any other ideas you have will help.

**Comment by MaKleSoft** [comment]
> I'm not 100% sure if the App Platform has the instances with the disks in write mode.

Yep, I think that's exactly it. AFAIK app platform "services" are all supposed to be stateless, so we either need to mount a "Volume" or provision a DB.

**Comment by BrunoBernardino** [comment]
Alright, seems I'll have to provision a DB because [they don't support "volumes" yet](https://www.digitalocean.com/community/questions/can-app-platform-persist-data-on-a-volume-between-releases), and an object storage won't really work, here.

**Comment by BrunoBernardino** [comment]
@MaKleSoft I'm leaving the `branch` set to `feature-digitalocean-one-click-install` so you can test, if you want.

File: packages/server/src/storage/postgres.ts [hunk 3]
```diff
@@ -73,9 +76,10 @@ export class PostgresStorage implements Storage {
     private _ensuredTables = new Map<string, Promise<void>>();
 
     constructor(public config: PostgresConfig) {
-        const { host, user, password, port, database, tls, tlsCAFile, tlsRejectUnauthorized } = config;
+        const { host, user, password, port, database, tls, tlsCAFile, tlsCAFileContents, tlsRejectUnauthorized } =
+            config;
         const tlsCAFilePath = tlsCAFile && resolve(process.cwd(), tlsCAFile);
-        const ca = tlsCAFilePath && readFileSync(tlsCAFilePath).toString();
+        const ca = (tlsCAFilePath && readFileSync(tlsCAFilePath).toString()) || tlsCAFileContents;
```
Comment (MaKleSoft) [inline packages/server/src/storage/postgres.ts:82]: I think `tlsCAFileContents` should take precedence of `tsCAFilePath`, so:

```suggestion
        const ca = tlsCAFileContents || (tlsCAFilePath && readFileSync(tlsCAFilePath).toString());
```

**Comment by MaKleSoft** [comment]
Getting this error when clicking the button:

Screenshot 2022-11-29 at 14 37 36

**Comment by BrunoBernardino** [comment]
@MaKleSoft are you using this URL: https://cloud.digitalocean.com/apps/new?repo=https://github.com/padloc/padloc/tree/feature-digitalocean-one-click-install ?

**Review by MaKleSoft — APPROVED** [review]
Alright, good stuff! Even set up a test instance and it works!

**Pull request merged**